internal/service/elasticache/replication_group.go

```diff
@@ -503,8 +508,23 @@ func resourceReplicationGroupCreate(ctx context.Context, d *schema.ResourceData,
 		input.PreferredCacheClusterAZs = flex.ExpandStringValueList(v.([]interface{}))
 	}
 
-	if v, ok := d.GetOk("replicas_per_node_group"); ok {
-		input.ReplicasPerNodeGroup = aws.Int32(int32(v.(int)))
+	rawConfig := d.GetRawConfig()
+	rawReplicasPerNodeGroup := rawConfig.GetAttr("replicas_per_node_group")
+	if rawReplicasPerNodeGroup.IsKnown() && !rawReplicasPerNodeGroup.IsNull() {
+		var v int32
+		err := gocty.FromCtyValue(rawReplicasPerNodeGroup, &v)
+		if err != nil {
+			path := cty.GetAttrPath("replicas_per_node_group")
+			diags = append(diags, errs.NewAttributeErrorDiagnostic(
+				path,
+				"Invalid Value",
+				"An unexpected error occurred while reading configuration values. "+
+					"This is always an error in the provider. "+
+					"Please report the following to the provider developer:\n\n"+
```

Comment:
I've seen this string in a number of places -- do we want to extract to a constant?

Reply:
Yes, I plan to do it as part of the ongoing AutoFlex stuff. It would be even better if `terraform-plugin-framework` exported it.